larpandoradlcontent/LArTwoDReco/DLThreeDClusterSplittingAlgorithm.cc

```diff
@@ -0,0 +1,1350 @@
+    IntVector &splitIndices) const
+{
+    IntVector filtered;
+
+    // Reject if near nu vertex
+    const CartesianVector &nuVertex3D(m_pNuVertexList->front()->GetPosition());
+    const CartesianVector nuVertexPosition(LArGeometryHelper::ProjectPosition(this->GetPandora(), nuVertex3D, hitType));
+
+    for (const int splitIndex : splitIndices)
+    {
+        if (splitIndex >= static_cast<int>(clusterPath.size()))
+            throw StatusCodeException(STATUS_CODE_FAILURE);
+
+        const CartesianVector &position(clusterPath.at(splitIndex).m_pHit->GetPositionVector());
+        
+        if ((nuVertexPosition - position).GetMagnitude() > m_minNuVertexSep)
+            filtered.push_back(splitIndex);
+    }
+
+    // Reject if near gap 
+    if (filtered.size() != splitIndices.size())
+    {
+        const FloatVector &gapSep(features.at("GapSep").m_sequence);
+
+        for (const int splitIndex : splitIndices)
+        {
+            if (std::find(filtered.begin(), filtered.end(), splitIndex) != filtered.end())
+                continue;
+
+            bool nearGap(false);
+            for (int offset = -m_gapSearchRegion; offset <= m_gapSearchRegion; ++offset)
+            {
+                int newIndex(splitIndex + offset);
+
+                // Gap indicated by 0.f
+                if ((newIndex >= 0) && (newIndex < int(gapSep.size())) && (gapSep.at(newIndex) < 0.5f))
+                {
+                    nearGap = true;
+                    break;
+                }
+            }
+
+            if (!nearGap)
+                filtered.push_back(splitIndex);
+        }
+    }
+
+    splitIndices.swap(filtered);
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+void DLThreeDClusterSplittingAlgorithm::ThreeViewMatching(const CartesianPointVector &splitPosU, const CartesianPointVector &splitPosV, 
+    const CartesianPointVector &splitPosW, IntVector &usedU, IntVector &usedV, IntVector &usedW)
+{
+    for (unsigned int iU = 0; iU < splitPosU.size(); ++iU)
+    {
+        if (std::find(usedU.begin(), usedU.end(), iU) != usedU.end())
+            continue;
+
+        for (unsigned int iV = 0; iV < splitPosV.size(); ++iV)
+        {
+            if (std::find(usedV.begin(), usedV.end(), iV) != usedV.end())
+                continue;
+                    
+            for (unsigned int iW = 0; iW < splitPosW.size(); ++iW)
+            {
+                if (std::find(usedW.begin(), usedW.end(), iW) != usedW.end())
+                    continue;
+                        
+                const float minX(std::min(splitPosU.at(iU).GetX(), std::min(splitPosV.at(iV).GetX(), splitPosW.at(iW).GetX())));
+                const float maxX(std::max(splitPosU.at(iU).GetX(), std::max(splitPosV.at(iV).GetX(), splitPosW.at(iW).GetX())));
+
+                if ((maxX - minX) > m_threeViewMatchMaxXSpan)
```

Comment:
Apopreciate this can be annoying, but how does this cope with wide hits? (probably not our main over-merging scenario?)